contracts/token-distribution/SaleRounds.sol

```diff
@@ -204,115 +159,186 @@ contract SaleRounds is TokenDistribution, GameOwner, ERC20 {
+    function reserveTokensInternal(RoundType _roundType, address _to, uint _amount) private {
+        require(roundDistribution[_roundType].supply >= _amount, "given amount is bigger than max supply for the round");
+        require(roundDistribution[_roundType].totalRemaining >= _amount, "total remaining round amount is not enough");
+        roundDistribution[_roundType].totalRemaining -= _amount;
+        roundDistribution[_roundType].startTime = block.timestamp;
+        reservedBalances[_roundType][_to] += _amount;
     }
 
-    function getTotalPending(string calldata _roundType, address _to) public view returns(uint256) {
-        RoundType roundType = getRoundTypeByKey(_roundType);
+    function initialReserveAndMint(address[] memory walletAddresses) private {
+        require(walletAddresses.length == 6, "walletAddresses array is not the correct length");
+        address exchangesWalletAddress = walletAddresses[0];
+        address playAndEarnWalletAddress = walletAddresses[1];
+        address socialWalletAddress = walletAddresses[2];
+        address teamWalletAddress = walletAddresses[3];
+        address treasuryWalletAddress = walletAddresses[4];
+        address advisorsWalletAddress = walletAddresses[5];
 
-        return reservedBalances[roundType][_to];
+        //ALLOCATIONS WITH WALLET CONSTANT
+        require(playAndEarnWalletAddress != address(0), "Play and earn wallet address is 0x0");
+        reserveTokensInternal(RoundType.PLAYANDEARN, playAndEarnWalletAddress, playAndEarnDistribution.supply);
+
+        require(socialWalletAddress != address(0), "Social wallet address is 0x0");
+        reserveTokensInternal(RoundType.SOCIAL, socialWalletAddress, socialDistribution.supply);
+
+        require(teamWalletAddress != address(0), "Team wallet address is 0x0");
+        reserveTokensInternal(RoundType.TEAM, teamWalletAddress, teamDistribution.supply);
+
+        require(treasuryWalletAddress != address(0), "Treasury wallet address is 0x0");
+        reserveTokensInternal(RoundType.TREASURY, treasuryWalletAddress, treasuryDistribution.supply);
+
+        require(advisorsWalletAddress != address(0), "Advisors wallet address is 0x0");
+        reserveTokensInternal(RoundType.ADVISOR, advisorsWalletAddress, advisorsDistribution.supply);
+
+        require(exchangesWalletAddress != address(0), "Exchanges wallet address is 0x0");
+
+        // Initial minting of 40% of total supply for exchanges distribution
+        uint256 initialExchangesSupply = 60_000_000;
+        _mint(exchangesWalletAddress, initialExchangesSupply);
+        roundDistribution[RoundType.EXCHANGES].totalRemaining -= initialExchangesSupply;
+        claimedBalances[RoundType.EXCHANGES][exchangesWalletAddress] += initialExchangesSupply;
+
+        // Reserving rest of the 60% of the supply for exchanges distribution
+        reserveTokensInternal(RoundType.EXCHANGES, exchangesWalletAddress, exchangesDistribution.supply - initialExchangesSupply);
```

Comment:
Where did this requirement come from? This is literally new for me.

Reply:
From the token allocation spreadsheet -> https://themoonholding.sharepoint.com/:x:/r/sites/MoonGaming-Leadership/_layouts/15/doc2.aspx?sourcedoc=%7B6cf160b8-2ce8-4a90-ae87-112427305155%7D&action=view&wdinitialsession=40254614-c1b2-42af-b6dd-0f461bd2b004&wdrldsc=5&wdrldc=2&wdrldr=BootTimeMismatch&cid=fb48221d-8f0f-4119-bace-6dd3aa7fb554